src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -943,6 +974,29 @@ def __chunked_iter_fill__(cls, parent, name, data, options=None):
                 dset.resize(new_shape)
             # Process and write the data
             dset[chunk_i.selection] = chunk_i.data
+        except StopIteration:
+            return False
+        return True
+
+    @classmethod
+    def __chunked_iter_fill__(cls, parent, name, data, options=None):
```

Comment:
Should this function be called somewhere? As far as I can see, it is not called anywhere, and could just be removed. But perhaps it should be called if exhaust_dci is false (line 875)?

Reply:
I originally took it out, but put it back for backward compatibility.

Reply:
Ok. Can you add a comment saying this is currently unused? Or remove it? Since it is a private function, I don't think it is needed for backwards compatibility. All of the tests of this function would have to be change though.

Reply:
It’s not private. Double underscores on one side makes it private. Double underscores on both sides makes it public. I want to keep it around so we still have a function for exhausting a DataChunkIterator

Reply:
Ultimately, I think it may be useful to have this configurable, i.e., be able to switch behavior via a flag to either do it round-robin or exhaust one-at-a-time. If you need to merge this, then I think it's fine to do that in a separate issue/pr.

Reply:
That was the intention of the exhaust_dci argument of write_dataset